### PR TITLE
Custom Norm Func for GCNConv

### PR DESCRIPTION
**Pull request by rbSparky**
Fixes issue [Issue #277](https://github.com/CarloLucibello/GraphNeuralNetworks.jl/issues/277)

- Added custom norm fn
- Changed tests

### CODE REVIEW

File: src/layers/conv.jl [hunk 1]
```diff
@@ -32,7 +32,7 @@ and optionally an edge weight vector.
 
 # Forward
 
-    (::GCNConv)(g::GNNGraph, x::AbstractMatrix, edge_weight = nothing) -> AbstractMatrix
+    (::GCNConv)(g::GNNGraph, x::AbstractMatrix, edge_weight = nothing, normalization_fn::Function = (d) -> 1 ./ sqrt.(d)) -> AbstractMatrix
```
Comment (CarloLucibello) [inline src/layers/conv.jl:35]:
```suggestion
    (::GCNConv)(g::GNNGraph, x::AbstractMatrix, edge_weight = nothing, norm_fn::Function = d -> 1 ./ sqrt.(d)) -> AbstractMatrix
```
I prefer a shorter name.

File: src/layers/conv.jl [hunk 2]
```diff
@@ -98,7 +99,8 @@ check_gcnconv_input(g::GNNGraph, edge_weight::Nothing) = nothing
 
 function (l::GCNConv)(g::GNNGraph, 
                       x::AbstractMatrix{T},
-                      edge_weight::EW = nothing
+                      edge_weight::EW = nothing,
+                      normalization_fn::Function = (d) -> 1 ./ sqrt.(d)
```
Comment (CarloLucibello) [inline src/layers/conv.jl:103]:
```suggestion
                      norm_fn::Function = d -> 1 ./ sqrt.(d)  
```

**Comment by CarloLucibello** [comment]
Thanks, this looks nice. The docstring should better specify how the norm function is used though.

**Comment by rbSparky** [comment]
> Thanks, this looks nice. The docstring should better specify how the norm function is used though.

Done

**Comment by CarloLucibello** [comment]
tests are not passing

**Comment by rbSparky** [comment]
> tests are not passing

Seems like in the ``Julia nightly`` release, there is some issue that seems to be unrelated to the change:

**ERROR**: The following 2 direct dependencies failed to precompile:
cuDNN [02a925ec-e4fe-4b08-9a7e-0d78e3d38ccd]

I suppose that it would be enough to get the ``Julia 1 - ubuntu-latest`` tests to pass?

**Pull request merged**